### PR TITLE
Fixed nil fields being dereferenced for public ips

### PR DESCRIPTION
**Pull request by thapr0digy**
When iterating through some VMs, the PublicIPAddress field on the ipConfig struct can be nil. In these cases, we don't want to dereference and just want to go to the next configuration. 

Also, the IPAddress for a nic IP could also be nil and in the same case, we do not want to dereference and add to the list. We'd rather leave it off.

### CODE REVIEW

File: pkg/providers/azure/vm.go [hunk 1]
```diff
@@ -51,6 +51,12 @@ func (d *vmProvider) GetResource(ctx context.Context) (*schema.Resources, error)
 				for _, ipConfig := range ipconfigList {
 					privateIP := *ipConfig.PrivateIPAddress
 
+					// The PublicIPAddress field can be nil especially for SQL Server
+					// VMs that are pulled.
+					// In these cases, we just want to return an empty error
+					if ipConfig.PublicIPAddress == nil {
+						return nil, nil
```
Comment (ehsandeep) [inline pkg/providers/azure/vm.go:58]: this should probably `continue`, instead of returning as we do want to process other vms in the loop

**Review by ehsandeep — CHANGES_REQUESTED** [review]

File: pkg/providers/azure/vm.go (hunk 1 above)
Comment (thapr0digy) [reply pkg/providers/azure/vm.go:58]: I had tried with continue, but never received results back. Not sure why that's occurring but when returning nil,nil I receive results back.

Comment (jgalindo-connectbase) [reply pkg/providers/azure/vm.go]: Resolved. Had to add another check later on for more nil

Comment (thapr0digy) [reply pkg/providers/azure/vm.go]: Resolved. Had to add another check for nil later on.

**Comment by thapr0digy** [comment]
@ehsandeep let me know if those changes look good for you.

File: pkg/providers/azure/vm.go [hunk 2]
```diff
@@ -61,6 +68,10 @@ func (d *vmProvider) GetResource(ctx context.Context) (*schema.Resources, error)
 						return nil, err
 					}
 
+					if publicIP.IPAddress == nil {
+						return nil, nil
```
Comment (Ice3man543) [inline pkg/providers/azure/vm.go:72]: Need continue here as well

**Review by Ice3man543 — CHANGES_REQUESTED** [review]

**Pull request merged**